src/Common/src/CoreLib/System/Globalization/DateTimeParse.cs

```diff
@@ -15,7 +15,15 @@ internal static class DateTimeParse
 
         internal delegate bool MatchNumberDelegate(ref __DTString str, int digitLen, out int result);
 
-        internal static MatchNumberDelegate m_hebrewNumberParser = new MatchNumberDelegate(DateTimeParse.MatchHebrewDigits);
+        internal static MatchNumberDelegate m_hebrewNumberParser;
+
+        static DateTimeParse()
+        {
+            if (!GlobalizationMode.Invariant)
+            {
+                m_hebrewNumberParser = new MatchNumberDelegate(DateTimeParse.MatchHebrewDigits);
```

Comment:
This should be LazyInitialized instead of explicit static cctor which we won't be able easily remove

Reply:
I  am fine with using some `Lazy<T>` initialization and / or putting that inside `DoStrictParse()`.

However, the field initializer that this code was previously using has to go away.

The general rule here is that the pattern `new SomeDelegate (ConditionalMethod)` must not be used outside a linker conditional (like for instance `GlobalizationMode.Invariant`), so you cannot put it into a field initializer.  The linker is unable to "go inside" that delegate initialization and we also don't have full support for dead field elimination yet.

Reply:
That's what I meant. You keep only `internal static MatchNumberDelegate m_hebrewNumberParser;` with `LazyInitializer.EnsureInitialized (ref m_hebrewNumberParser, ...)` inside `DoStrictParse`

Reply:
Like this:
```
            if (!GlobalizationMode.Invariant && (CalendarId)parseInfo.calendar.ID == CalendarId.HEBREW)
            {
                LazyInitializer.EnsureInitialized (ref m_hebrewNumberParser, () => new MatchNumberDelegate(DateTimeParse.MatchHebrewDigits));
                parseInfo.parseNumberDelegate = m_hebrewNumberParser;
                parseInfo.fCustomNumberParser = true;
            }
```

or do we need locking?

Reply:
yep, EnsureInitialized is thread-safe

Reply:
Okay, done.